scenario/uploaddata.go

```diff
@@ -70,124 +75,103 @@ func (settings UploadDataSettings) Execute(
 		return
 	}
 
-	restHandler := sessionState.Rest
+	sessionState.Rest.GetAsync(
+		fmt.Sprintf("%s/%s/quota", host, datafileEndpoint), actionState, sessionState.LogEntry, nil,
+	)
 
 	dataConnectionID, err := sessionState.FetchDataConnectionID(actionState, host, settings.SpaceID)
 	if err != nil {
 		actionState.AddErrors(errors.WithStack(err))
 		return
 	}
 
-	dataFiles, err := sessionState.FetchQixDataFiles(actionState, host, dataConnectionID)
+	fileName := filepath.Base(settings.Filename)
+
+	existingFile, err := sessionState.FetchQixDataFile(actionState, host, dataConnectionID, fileName)
 	if err != nil {
 		actionState.AddErrors(errors.WithStack(err))
 		return
 	}
 
-	fileName := filepath.Base(settings.Filename)
-
-	var existingFile *elasticstructs.QixDataFile
-	// check to see if file exists already
-	for _, file := range dataFiles {
-		if file.Name == fileName {
-			existingFile = &file
-			break
-		}
-	}
-
 	if existingFile != nil && !settings.Replace {
 		sessionState.LogEntry.Logf(
-			logger.WarningLevel, "data file not uploaded, filename<%s> already exists and replace set to false", settings.Filename,
+			logger.WarningLevel, "data file not uploaded, filename<%s> already exists and replace set to false", fileName,
 		)
 		sessionState.Wait(actionState)
 		return
 	}
 
-	file, err := os.Open(settings.Filename)
+	fileContent, err := ioutil.ReadFile(settings.Filename)
 	if err != nil {
 		actionState.AddErrors(errors.Wrapf(err, "failed to open file <%s>", settings.Filename))
 		return
 	}
-	defer func() {
-		err = file.Close()
-		if err != nil {
-			actionState.AddErrors(errors.Wrapf(err, "failed to close file <%s>", settings.Filename))
-		}
-	}()
-
-	body := helpers.GlobalBufferPool.Get()
-	defer helpers.GlobalBufferPool.Put(body)
-	writer := multipart.NewWriter(body)
+	fileSize := len(fileContent)
+
+	tempContentPostRequest := sessionState.Rest.PostWithHeadersAsync(
+		fmt.Sprintf("%s/%s", host, tempContentFilesEndpoint),
+		actionState, sessionState.LogEntry, nil, map[string]string{
+			"upload-length": strconv.Itoa(fileSize),
+			"tus-resumable": "1.0.0",
+		}, &session.ReqOptions{
```

Comment:
this looks like it should be uploaded using TUS? I.e. being able to automatically resume the upload on errors midstream etc. Have  a look at upload app where we use a library for TUS uploads.

Reply:
Most definitely. Checking out upload app 👍